buildroot/share/PlatformIO/scripts/common-dependencies.py

```diff
@@ -16,6 +16,28 @@
 	# PIO >= 4.4
 	from platformio.package.meta import PackageSpec as PackageManager
 
+PIO_VERSION_MIN = (5, 0, 3)
+try:
+	from platformio import VERSION as PIO_VERSION
+	weights = (1000, 100, 1)
+	version_min = sum([x[0] * float(re.sub(r'[^0-9]', '.', str(x[1]))) for x in zip(weights, PIO_VERSION_MIN)])
+	version_cur = sum([x[0] * float(re.sub(r'[^0-9]', '.', str(x[1]))) for x in zip(weights, PIO_VERSION)])
+	if version_cur < version_min:
+		print()
+		print("**************************************************")
+		print("******      An update to PlatformIO is      ******")
+		print("******  required to build Marlin Firmware.  ******")
+		print("******                                      ******")
+		print("******      Minimum version: ", PIO_VERSION_MIN, "    ******")
+		print("******      Current Version: ", PIO_VERSION, "    ******")
+		print("******                                      ******")
+		print("******   Update PlatformIO and try again.   ******")
+		print("**************************************************")
+		print()
+		exit(1)
+except:
+	print("Can't detect PlatformIO Version")
```

Comment:
This exit(1) isn't working for me, because it actually raises a SystemExit exception that is then caught.
I will probably just push a minor change to fix that.

So I see this, followed by my build:

```
**************************************************
******      An update to PlatformIO is      ******
******  required to build Marlin Firmware.  ******
******                                      ******
******      Minimum version:  (5, 0, 4)     ******
******      Current Version:  (5, 0, 3)     ******
******                                      ******
******   Update PlatformIO and try again.   ******
**************************************************

Can't detect PlatformIO Version
```

Reply:
Thanks. I did that last change to avoid breaking with weird version number, but did run again with new version number to try.